bip-0174.mediawiki

```diff
@@ -133,8 +133,8 @@ The currently defined global types are as follows:
 *** <tt>{32-bit uint}</tt>
 
 * Type: Proprietary Use Type <tt>PSBT_GLOBAL_PROPRIETARY = 0xFC</tt>
-** Key: Variable length identifier prefix, followed by a subtype, followed by the key data itself.
-*** <tt>{0xFC}|<prefix>|{subtype}|{key data}</tt>
+** Key: Compact size unsigned integer, followed by identifier prefix of that length, followed by a subtype, followed by the key data itself.
```

Comment:
Because everything variable length in Bitcoin is prefixed by a compact size unsigned int, "Variable length ..." is the correct way to describe this. Further clarification of this should be in a parenthetical rather than listed as its own separate entity.

Reply:
This isn't even true in this BIP.

The key data is variable length.  It is not prefixed by a compact size unsigned int!

Reply:
> The key data is variable length. It is not prefixed by a compact size unsigned int!

Uhh... no? It definitely is prefixed by the size, it's just usually `0x01`.

Reply:
Not quite: the *entire key* is prefixed by the size.  The keydata length is *implied*.

You also use 0 as a terminator in various places in this BIP, eschewing explicit length there.

And bitcoin uses NUL as a terminator on command names in the wire protocol (within a fixed length field, because reasons?)

Given this lack of consistency, I think we're best off being horribly explicit?